### PR TITLE
[Google Blockly] Enable behavior editor (part 1)

### PR DESCRIPTION
**Pull request by molly-moen**
Enable the behavior editor for Google Blockly. This PR does not cover adding the `this sprite` toolbox, because work is being done to enable that in [this PR](https://github.com/code-dot-org/code-dot-org/pull/54080). It covers the following:
- If the modal function editor is enabled and the experiment is turned on, we add an edit button to behavior call blocks when they are in the workspace. Users can click that edit button to edit the behaviors, similar to functions.
- Behaviors will render `with: this sprite` at the end of the function definition block, the same as when they were on the main workspace.
- Clicking `Create a behavior` will open the modal editor, if the experiment is enabled.
- Behaviors are now serialized with descriptions.

## Links

- jira ticket: [CT-108](https://codedotorg.atlassian.net/browse/CT-108)

## Testing story
Tested locally. This change only impacts Google Blockly labs with the modal function editor experiment enabled.

## Follow-up work
After the linked PR (above) is in, we need to make sure the mini toolbox is populated with `this sprite` correctly for behaviors. We also have a [follow up task](https://codedotorg.atlassian.net/browse/CT-121) to handle descriptions for both functions and behaviors once that PR is in.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js [hunk 1]
```diff
@@ -75,11 +76,12 @@ export const blocks = GoogleBlockly.common.createBlockDefinitionsFromJsonArray([
       'procedure_defnoreturn_set_comment_helper',
       'procedure_def_set_no_return_helper',
       'behaviors_block_frame',
+      'modal_procedures_no_destroy',
```
Comment (ebeastlake) [inline apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js:79]: This keeps the behavior definition block in the modal editor from being deleted?

Comment (ebeastlake) [reply apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js:79]: Update: I wondered why this didn't already exist, but it looks like we needed to add this when we added the procedure map, so assuming this all sounds right, no further questions!

**Review by ebeastlake — APPROVED** [review]

**Comment by ebeastlake** [comment]
Nice work!

File: apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js:79]: It was something I had to add for the [procedures](https://github.com/code-dot-org/code-dot-org/blob/f4896810ef156cbed830fab30f2aad0b88769022/apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.js#L166) once we were using the modal function editor. It prevents the procedure from being deleted from the procedure map for the editor workspace when we clear the editor workspace.

**Pull request merged**